pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/JavaInstanceMain.java

```diff
@@ -195,12 +213,13 @@ public void run() {
             }
         });
 
-        // registering jvm metrics to prometheus
-        DefaultExports.initialize();
-
         log.info("Starting runtimeSpawner");
         runtimeSpawner.start();
 
+        // starting metrics server
+        log.info("Starting metrics server on port {}", metrics_port);
```

Comment:
start only if metrics port is specified? Same for python

Reply:
Are you saying we should have some sort of flag to determine whether we should run the http server for metrics?

Reply:
No, just that a negative port value means not to start metrics server

Reply:
are we exposing that config to the user somehow? Because currently its always going to be set